engine/src/flutter/impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -546,9 +546,8 @@ std::shared_ptr<CommandBuffer> ContextVK::CreateCommandBuffer() const {
     DescriptorPoolMap::iterator current_pool =
         cached_descriptor_pool_.find(std::this_thread::get_id());
     if (current_pool == cached_descriptor_pool_.end()) {
-      descriptor_pool =
-          (cached_descriptor_pool_[std::this_thread::get_id()] =
-               std::make_shared<DescriptorPoolVK>(weak_from_this()));
+      descriptor_pool = (cached_descriptor_pool_[std::this_thread::get_id()] =
+                             descriptor_pool_recycler_->GetDescriptorPool());
```

Comment:
Here we are doing a lookup based on what thread we are on.  Can we expand that to include the recycler too so we don't have to introduce mutexes?

Reply:
This change doesn't introduce any new mutexes